test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -374,6 +374,46 @@ TEST_F(testNuma, checkModeInterleave) {
     numa_bitmask_free(retrieved_nodemask);
 }
 
+// Test for allocations on numa nodes with interleave mode enabled and custom part size set.
+// The page allocations are interleaved across the set of nodes specified in nodemask.
+TEST_F(testNuma, checkModeInterleaveCustomPartSize) {
+    constexpr int part_num = 1024;
+    size_t page_size = sysconf(_SC_PAGE_SIZE);
+    size_t part_size = page_size * 1;
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+
+    std::vector<unsigned> numa_nodes = get_available_numa_nodes();
+    set_all_available_nodemask_bits(nodemask);
+
+    os_memory_provider_params.numa_list = numa_nodes.data();
+    os_memory_provider_params.numa_list_len = numa_nodes.size();
+    os_memory_provider_params.numa_mode = UMF_NUMA_MODE_INTERLEAVE;
+    os_memory_provider_params.part_size = part_size;
+    initOsProvider(os_memory_provider_params);
+
+    umf_result_t umf_result;
+    umf_result = umfMemoryProviderAlloc(os_memory_provider,
```

Comment:
Could you extend this to also test sizes that are:
- `< part_size`
- `> part_size` but not a multiple of part_size

Reply:
Done